posydon/popsyn/synthetic_population.py

```diff
@@ -239,22 +288,41 @@ def parse(self, S1_state=None, S2_state=None, binary_state=None,
                 if any(sel):
                     df_tmp = pd.DataFrame()
                     df_tmp = df.loc[sel]
-                    sel_met.extend(sel)
                     # store metallicity
                     df_tmp['metallicity'] = met
                     # concatenate results
                     df_sel_met = pd.concat([df_sel_met, df_tmp])
                     del df_tmp
 
+            # check last binary if it should be included
+            if last_binary_df is not None:
+                logic = self.apply_logic(last_binary_df, 
+                                    S1_state     = S1_state,
+                                    S2_state     = S2_state,
+                                    binary_state = binary_state,
+                                    binary_even  = binary_event,
+                                    step_name    = step_name,
+                                    invert_S1S2  = invert_S1S2)
+                
+                # The last binary is selected
+                if any(logic) == True:
+                    df_tmp = last_binary_df.loc[logic]
+                    df_sel_met = pd.concat([df_sel_met, df_tmp])
+
+            # get unique indicies
+            sel_met = df_sel_met.index.unique()
+            
             if k > 0 and df_sel.shape[0] > 0:
                 shift_index = max(np.unique(df_sel.index)) + 1 
```

Comment:
Wouldn't it be better to always shift by the maximum index of all the data instead of only using the selection?
I have in mind to make two sections from the same population, than the same systems could get a different index at the moment.